source/usage-examples/findOne.txt

```diff
@@ -3,3 +3,93 @@ Find a Document
+
+..
+   options
+
+You can specify additional options when retrieving a document such as whether to
+**sort** the matches of your query, which retrieves the matching document
+with the minimum or maximum value as specified in your sort criteria. You can
+also specify a **projection**, which configures the fields in the returned
+document. You can specify additional options by passing the ``FindOne()`` method
+a pointer to an instance of a ``FindOneOptions`` value. You can configure a
+``FindOneOptions`` value by using setter methods such as the
+``*FindOneOptions.SetProjection()`` method.
+
+..
+   return
+
+If the query filter you specified in your ``FindOne()`` method call matches one
+or more documents in the collection, it returns the first matched result. The
+return type ``SingleResult`` represents a single document returned from a query
+to a MongoDB instance. You can call the ``*SingleResult.Decode()`` method
+to unmarshall a ``SingleResult`` value into a variable. Pass the ``Decode()`` 
+method a pointer to the variable that you would like to contain your returned
+document. 
+
+..
+   error 
+
+The ``Decode()`` method will return the error value of your ``FindOne()`` method
+call. 
+
+- If a document matched your query, and there are no errors retrieving the
+  document, the ``Decode()`` method will return ``nil``. 
+- If no documents matched your query, the ``Decode()`` method will return an
+  ``ErrNoDocuments`` error. 
+- If there was an error retrieving your document, the ``Decode()`` method will
+  return the error. 
+
+Example
+-------
+
+The following code example retrieves a single document from the ``movies``
+collection. It uses the following arguments for the ``FindOne()`` method:
+
+- An empty **Context** instance returned from the ``context.TODO()`` function.
+- A **query document** that only matches movies with the title exactly matching
+  the text ``"The Room"``.
+- A **configuration** that specifies a **sort** on matched documents in
+  descending order by rating, such that the retrieved document has the highest
+  rating. Our configuration also specifies a **projection** for our results so
+  that our MongoDB instance only sends the ``title`` and ``imdb`` fields of
+  our retrieved document.
+
+.. include:: /includes/connect-guide-note.rst
+
+.. literalinclude:: findOne.go
+   :language: go
+
+If you run the example above, you should see a result that looks like this:
+
+.. literalinclude:: findOne.go.txt
+   :language: none
+
+For more information on the methods, interfaces, structs, and functions
+mentioned on this page, see the following API documentation:
+
+- :go-api:`*Collection.FindOne() </mongo#Collection.FindOne>`
+- `Context <https://pkg.go.dev/context#Context>`__
+- :go-api:`FindOneOptions </mongo/options#FindOneOptions>`
+- :go-api:`*FindOneOptions.SetProjection() </mongo/options#FindOneOptions.SetProjection>`
+- :go-api:`options.FindOne() </mongo/options#FindOne>`
```

Comment:
What's different between FindOne and FindOneOptions that made you include the package name? If it's to differentiate the namespace from Collection.FindOne(), maybe you could consistently include the package name? Or is there some other criteria?

Reply:
The criteria in the list that I tried to follow were:

structs: do not include package name
methods: include struct name
functions: include package name

I am in no way attached to these criteria. My thinking was, "there is not a convention in the docs for go package names, and I can't find a convention elsewhere, so I should take a stab at making one and it will be discussed in the review phase". I am now realizing I should have brought this up explicitly, possibly as a note in the initial PR. I apologies for not drawing your attention to this immediately. Going forward, I will try to do a better job of identifying these potential far reaching decisions and brining them up for discussion.

[`FindOneOptions`](https://pkg.go.dev/github.com/mongodb/mongo-go-driver/mongo/options#FindOneOptions) is a struct type where as [`options.FindOne()`](https://pkg.go.dev/github.com/mongodb/mongo-go-driver/mongo/options#FindOne) is a function. I am not sure of the best convention for including package/struct names. Here are some options:

1. Include the entire package and or struct path for everything
2. Include the package path for functions and structs, and include the struct path for methods
3. Criteria listed above

After writing out this comment, I think I support option 1. What is your preference?